### ISSUE TITLE
console: autodeploy logs take a long time to stream or don't come through at all

### ISSUE THREAD

**Issue by nguyendon** [anomalyco/sst#5196]
Two deploy examples:

- `l0dd1h1esbohravcn2arjbw9`
- `u1kzteihkarv2h6kdm6p51rw`
 
<img width="518" alt="image" src="https://github.com/user-attachments/assets/443f2bc8-da3d-4ea6-92aa-518ad76f8796">

**Comment by SantiagoPaak** [comment]
It's happening to me too. I have a failed autodeploy and a failed redeploy and logs aren't coming through after 15 minutes.

Unsure if it's related to the problem, but the networking tab appears to make constant calls to `https://api.console.sst.dev/log/aws/scan` and succeeding, but getting empty data over and over.

**Comment by jayair** [comment]
We'll take a look.

**Comment by imharrisonking** [comment]
Hi all, I'm also having this issue with all of my autodeploys. Deploying locally is working fine though.
![image](https://github.com/user-attachments/assets/4d7bd006-cd31-400e-8f25-91829b29e11c)

**Comment by ben-sommer** [comment]
Exact same issue here, although my error message is `Failed to run: sst deploy --stage production`.

**Comment by ben-sommer** [comment]
You can view the logs directly in the AWS Console by going to CodeBuild in `us-east-1`.

For me the issue was with trying to build a Remix project using Vite. Because I had a monorepo structure, the build command did not install the dependencies that sub-projects had defined. Configuring workspaces in `package.json` resolved the issue.

**Comment by jayair** [comment]
@imharrisonking are you still having this issue?

**Comment by ben-sommer** [comment]
To clarify, I didn't fix the logs not coming through. These still don't show. I fixed the error stopping the build pipeline though.

**Comment by nilsingwersen** [comment]
Failed to run: /root/repo/node_modules/.bin/sst deploy

I got the same issue. The only thing that is special for my setup is that I'm using the cloudflare provider for my domain.
And I'm on europe-central-1

I even updated sst to 3.6.35

This is my repo: https://github.com/nilsingwersen/next-sst

Anything more I can provide?

**Comment by jayair** [comment]
@ben-sommer got it

@nilsingwersen is your issue that the logs are not coming through?

**Comment by nilsingwersen** [comment]
> [@ben-sommer](https://github.com/ben-sommer) got it
> 
> [@nilsingwersen](https://github.com/nilsingwersen) is your issue that the logs are not coming through?

My issued is that the autodeploy doesn't work for me and I don't get logs. I deleted my stack and created a new one and now it seems like I get logs but the deploy still doesn't work. Now I get this error:

sdk-v2/provider2.go:515: sdk.helper_schema: expected DNS record to not already be present but already exists: provider=cloudflare@5.49.0


Should I keep this here or create a new issue?

**Comment by jayair** [comment]
Yeah let's do a new issue and share the logs and maybe some details about what you have in your app.

**Comment by fwang** [comment]
For anyone whose AutoDeploy logs do not show up, can you send me the link to the build? (ie. https://console.sst.dev/xxxx/xxxx/autodeploy/xxxxxxxxxxxxxxxxxxxxx)

**Comment by imharrisonking** [comment]
Hi @jayair @fwang thanks for getting back. I noticed a more detailed error message in the console saying SST didn't have permission to create resources: 

`We don't have the permissions to create the necessary resources in your AWS account. You can fix this by creating these manually and trying again.`

So I did some digging in CloudFormation and seems like the issue is due to my AWS Lambda Concurrent Executions applied account-level quota being set to 10 instead of the default 1000. Weird and not sure why it's been set so low but will try to increase this quota and see if it solves the problem.

<img width="1439" alt="Image" src="https://github.com/user-attachments/assets/23b3a594-8a32-46fd-8e17-2f664070938d" />

**Comment by fwang** [comment]
Following up on this, can someone confirm if the build log show up in AWS CodeBuild console?

Also please send me a link to the build in SST console.

**Comment by nguyendon** [comment]
@fwang here is one that logs don't show up show up: https://console.sst.dev/libum/portal/autodeploy/najeqmagp7ev551ii2zxd42m

We have been able to view the build logs in AWS CodeBuild which is how we've been debugging builds. I haven't deployed in project in a while so not sure if there is another reason these logs aren't showing up here.

**Comment by fwang** [comment]
@nguyendon in ur case, the build above deployed the `prod` stage to the AWS account ending in `6865`, which has been disconnected from Console. And your current `prod` stage is in AWS account ending in `8238`. So the Console no longer has access to your CodeBuild logs.

Showing the logs is loading is confusing. We should display a better message instead.

Let me know if that makes sense, and if there's another build u want me to look into.

**Comment by nguyendon** [comment]
@fwang thats a pretty old build so I can't remember if it was one of the ones that was originally having issues. The AWS account ending in `6865` should have never been connected to this workspace and the build should have never been deployed to that account. They were never connected on our end, could this be a bug somehow?

**Comment by fwang** [comment]
The AWS account ending in `6865` was connected at `2024-11-22 08:12:49`, and disconnected after 3 minutes at `2024-11-22 08:15:13`. There's actually no way for Console to know about an AWS account unless it's manually connected.

**Comment by fwang** [comment]
Just pushed out a fix. The log page should be working now!

My bad, it was a bug on the frontend. The build was indeed deployed to the correct AWS account ending in `8238`. I got confused.

That said, the AWS account ending in `6865` was indeed connected to the `libum` workspace on `2024-11-22 08:12:49`, and disconnected right after at `2024-11-22 08:15:13`. The Console actually has no way to know about an AWS account unless it's manually connected.

**Comment by fwang** [comment]
This issue is fixed. Feel free to reopen if anyone still has issue.

**Comment by imharrisonking** [comment]
Hey @fwang thanks for this. Sorry for not getting back to you sooner, but I've been away! My logs are coming through now, one thing I'm having issues with still is the autodeploy is failing to deploy to us-east-1 as I'm already deploying that stage to eu-west-2 (where my IAM identity centre is).

Am I missing something or doing something wrong in terms of config? Do my stages have to be deployed to us-east-1 for autodeploy to work? Happy to create a new ticket for this if it's not a quick fix. Cheers!

<img width="1700" alt="Image" src="https://github.com/user-attachments/assets/e5ad690a-e5cd-4c11-9c00-bc02d1edca67" />

**Comment by jayair** [comment]
Yeah create a new issue.

**Comment by imharrisonking** [comment]
I redeployed my stages to us-east-1 which seems to have solved the issue, happy with this as a workaround
